src/main/java/ui/UpdateProgressWindow.java

```diff
@@ -0,0 +1,139 @@
+    public void hideWindow() {
+        Platform.runLater(window::hide);
+    }
+
+    /**
+     * Initialize the progress window
+     */
+    private void initProgressWindow() {
+        Stage stage = new Stage();
+        stage.setTitle(WINDOW_TITLE);
+
+        windowMainLayout = new VBox();
+
+        Group root = new Group();
+        Scene scene = new Scene(root);
+        stage.setScene(scene);
+
+        scene.setRoot(windowMainLayout);
+
+        window = stage;
+
+        reloadWindowLayout();
+    }
+
+    /**
+     * Create a new download progress bar
+     *
+     * @param downloadUrl url of download as identifier to the progress bar
+     * @param downloadLabel label of progress bar
+     * @return the method to set the progress to the progress bar of the download progress tracker
+     */
+    public Consumer<Double> createNewDownloadProgressBar(URL downloadUrl, String downloadLabel) {
+        ProgressBar progressBar = new ProgressBar(-1.0);
+        progressBar.setPrefWidth(400);
+        LabeledDownloadProgressBar downloadProgressBar =
+                new LabeledDownloadProgressBar(downloadUrl, downloadLabel, progressBar);
+
+        downloads.add(downloadProgressBar);
+
+        reloadWindowLayout();
+
+        return downloadProgressBar::setProgress;
+    }
+
+    /**
+     * Remove a download progress bar from update progress window
+     *
+     * This should be called after a download is completed
+     *
+     * @param downloadUrl download URL to be used as download identifier
+     */
+    public void removeDownloadProgressBar(URL downloadUrl) {
+        downloads.removeIf(dpt -> dpt.getUrlIdentifier() == downloadUrl);
+
+        reloadWindowLayout();
+    }
+
+    /**
+     * Update the update progress window's layout if there is any changes to download progress to be shown
+     */
+    private void reloadWindowLayout() {
+        assert window != null && windowMainLayout != null;
+
+        Platform.runLater(() -> {
+            windowMainLayout.getChildren().clear();
+
+            if (downloads.isEmpty()) {
+                Label noDownloadLabel = new Label();
+                noDownloadLabel.setText(LABEL_NO_DOWNLOAD);
+                noDownloadLabel.setPadding(new Insets(50));
+
+                windowMainLayout.getChildren().add(noDownloadLabel);
+            } else {
+                for (LabeledDownloadProgressBar progressTracker : downloads) {
+                    Label downloadLabel = new Label();
+                    downloadLabel.setText(progressTracker.getDownloadLabel());
+
+                    ProgressBar progressBar = progressTracker.getProgressBar();
+
+                    VBox downloadProgressItem = new VBox();
+                    downloadProgressItem.setSpacing(20);
+                    downloadProgressItem.setPadding(new Insets(20));
+                    downloadProgressItem.setAlignment(Pos.CENTER_LEFT);
+                    downloadProgressItem.getChildren().addAll(downloadLabel, progressBar);
```

Comment:
Maybe we could factor out the parts which create JavaFX nodes here? A bit more SLAP.